src/inc/til/string.h

```diff
@@ -104,51 +105,55 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
 #pragma warning(disable : 26481) // Don't use pointer arithmetic. Use span instead
         auto ptr = str.data();
         const auto end = ptr + str.length();
-        unsigned long base = 10;
         unsigned long accumulator = 0;
         unsigned long value = ULONG_MAX;
 
-        if (str.length() > 1 && *ptr == L'0')
+        if (!base)
         {
-            base = 8;
-            ptr++;
+            base = 10;
 
-            if (str.length() > 2 && (*ptr == L'x' || *ptr == L'X'))
+            if (str.length() > 1 && *ptr == '0')
             {
-                base = 16;
-                ptr++;
+                base = 8;
+                ++ptr;
+
+                if (str.length() > 2 && (*ptr == 'x' || *ptr == 'X'))
+                {
+                    base = 16;
+                    ++ptr;

```

Comment:
My IDE was complaining that the iterator result is discarded (`ptr++` returns a copy of the previous iterable state). I just changed it - can't hurt after all.